watertap/examples/flowsheets/ion_exchange/tests/test_ion_exchange_demo.py

```diff
@@ -0,0 +1,210 @@
+    def test_specific_operating_conditions(self, ix_0D):
+        m = ix_0D
+        ixf.set_operating_conditions(m, feed_mass_frac=feed_mass_frac)
+        ixf.initialize_system(m)
+        assert degrees_of_freedom(m) == 0
+
+        solver = get_solver()
+        results = solver.solve(m)
+        assert_optimal_termination(results)
+        assert value(m.fs.feed.properties[0].flow_vol_phase["Liq"]) == pytest.approx(
+            0.05, abs=1e-6
+        )
+        assert value(m.fs.product.properties[0].flow_vol_phase["Liq"]) == pytest.approx(
+            0.049901566, abs=1e-6
+        )
+        assert value(m.fs.regen.properties[0].flow_vol_phase["Liq"]) == pytest.approx(
+            0.016666, abs=1e-6
+        )
+        assert value(
+            sum(
+                m.fs.feed.properties[0].conc_equiv_phase_comp["Liq", j]
+                for j in m.fs.properties.ion_set
+            )
+        ) == pytest.approx(5, rel=1e-3)
+        assert value(
+            sum(
+                m.fs.product.properties[0].conc_equiv_phase_comp["Liq", j]
+                for j in m.fs.properties.ion_set
+            )
+        ) == pytest.approx(0.01057086, rel=1e-3)
+        assert value(
+            sum(
+                m.fs.regen.properties[0].conc_equiv_phase_comp["Liq", j]
+                for j in m.fs.properties.ion_set
+            )
+        ) == pytest.approx(471.99797, rel=1e-3)
+
+        assert value(m.fs.ion_exchange.number_columns) == pytest.approx(4, rel=1e-6)
+        assert value(m.fs.ion_exchange.bed_depth) == pytest.approx(1.7, rel=1e-6)
+        assert value(m.fs.ion_exchange.t_breakthru) == pytest.approx(
+            56759.75759, rel=1e-6
+        )
+        assert value(m.fs.ion_exchange.partition_ratio) == pytest.approx(
+            235.99899, rel=1e-6
+        )
+
+        assert value(m.fs.costing.specific_energy_consumption) == pytest.approx(
+            0.057352, abs=0.001
+        )
+        assert value(m.fs.costing.LCOW) == pytest.approx(0.219, abs=0.01)
+
+    @pytest.mark.requires_idaes_solver
+    @pytest.mark.component
+    def test_optimization(self, ix_0D):
+        m = ix_0D
+        ixf.optimize_system(m)
+        isinstance(m.fs.obj, Objective)
+        assert m.fs.obj.expr == m.fs.costing.LCOW
+        assert degrees_of_freedom(m) == 2
+
+        assert value(m.fs.feed.properties[0].flow_vol_phase["Liq"]) == pytest.approx(
+            0.05, abs=1e-6
+        )
+        assert value(m.fs.product.properties[0].flow_vol_phase["Liq"]) == pytest.approx(
+            0.049901566, abs=1e-6
+        )
+        assert value(m.fs.regen.properties[0].flow_vol_phase["Liq"]) == pytest.approx(
+            0.016666, abs=1e-6
+        )
+        ix = m.fs.ion_exchange
+        num_col = np.ceil(
+            ix.number_columns()
+        )  # To eliminate fractional number of columns
+        bed_depth = ix.bed_depth()
+        ix.bed_depth.fix(bed_depth)
+        ix.number_columns.fix(num_col)
```

Comment:
Did you ensure the ceil of the solution is truly the optimum and no weird concavity in the optimization exists? Without pure integer optimization, if the optimization solved to 4.2, 4 could actually be more optimum than 5. This may be possible with the objective function as LCOW, although not possible for something like minimizing a sizing parameter.

Reply:
I have done no real statistics based checking on this outside of some I made during the initial model development that showed minimal impact to LCOW when using floor or ceil, and I figured this was acceptable given the lack of integer optimization for our solver (and that I would leave it to the user to pick one or the other, or neither if they are mostly looking for costing results rather than design and so would be ok with fractional column values). 

I also found that the model is generally pushing the bounds of the column sizing variables and using floor can lead to less reliable solves because you are trying to cram the same amount of resin into fewer columns, thus the columns must be bigger. So I always use ceil... but yeah the design aspect of this model is not ideal for that reason.

Reply:
#885 is related -- where we have a parameter we might want to be an integer variable in certain contexts.

Reply:
Yea, this is definitely larger than the scope of this. I think this is fine to pass and have #885 to capture this for the future.